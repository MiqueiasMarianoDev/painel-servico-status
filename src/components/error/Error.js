import React from 'react'

export const Error = ({mensagem}) => {
    return (
      <div className="erro">
        <p>Oops, ocorreu um erro:</p>
        <p>{mensagem}</p>
      </div>
    );
  
}