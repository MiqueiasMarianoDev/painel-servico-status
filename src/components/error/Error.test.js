import React from 'react';
import { render, screen } from '@testing-library/react';
import { Error } from './Error';

describe('Error', () => {

  test('renders error message', () => {
    render(<Error mensagem="Erro de conexão" />);
    
    const errorMessage = screen.getByText(/Erro de conexão/i);
    expect(errorMessage).toBeInTheDocument();
  });

  test('renders default error message', () => {
    render(<Error />);
    
    const errorMessage = screen.getByText(/Oops, ocorreu um erro/i);
    expect(errorMessage).toBeInTheDocument();
  });


  test('snapshot', () => {
    const {asFragment} = render(<Error />);
    
    expect(asFragment()).toMatchSnapshot();
  });

});