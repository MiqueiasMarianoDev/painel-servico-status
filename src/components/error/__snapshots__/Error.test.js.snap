// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error snapshot 1`] = `
<DocumentFragment>
  <div
    class="erro"
  >
    <p>
      Oops, ocorreu um erro:
    </p>
    <p />
  </div>
</DocumentFragment>
`;
